File: dart/1893.dart
class Solution {
    bool isCovered(List<List<int>> j, int l, int r) {
        List<int> f = [];
        for (int i = 0; i < 52; i++)
            f.add(0);
        for (List<int> i in j) {
            f[i[0]]++;
            f[i[1] + 1]--;
        }
        int p = 0;
        for (int i = 0; i <= 51; i++) {
            p += f[i];
            if (i >= l && i <= r && p == 0)
                return false;
        }
        return true;
    }
}

File: python3/1893.py
class Solution:
    def isCovered(self, j: List[List[int]], l: int, r: int) -> bool:
        f = [0] * 52
        for i in j:
            f[i[0]] += 1
            f[i[1] + 1] -= 1
        p = 0
        for i in range(52):
            p += f[i]
            if i >= l and i <= r and p == 0:
                return False
        return True
